File: BackEnd/src/main/java/com/sansantek/sansanmulmul/crew/repository/CrewRepository.java
package com.sansantek.sansanmulmul.crew.repository;

import com.sansantek.sansanmulmul.crew.domain.Crew;
import jakarta.persistence.EntityManager;
import lombok.RequiredArgsConstructor;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface CrewRepository extends JpaRepository<Crew, Integer> {
    Optional<Crew> findByCrewId(int crewId);
    boolean existsByCrewIdAndLeader_UserId(int crewId, int userId);
    List<Crew> findAllByOrderByCrewCreatedAtDesc();
}

File: FrontEnd/app/src/main/java/com/sansantek/sansanmulmul/data/repository/CrewRepository.kt
package com.sansantek.sansanmulmul.data.repository

import com.sansantek.sansanmulmul.data.model.CrewMountainDetail
import com.sansantek.sansanmulmul.ui.util.RetrofiltUtil

class CrewRepository {
  suspend fun getCrewMountainDetail(crewId: Int): CrewMountainDetail? {
    return try {
      val response = RetrofiltUtil.crewService.getCrewMountainDetail(crewId)
      if (response.isSuccessful) {
        response.body()
      } else {
        null
      }
    } catch (e: Exception) {
      null
    }
  }
}